kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveStores.java

```diff
@@ -0,0 +1,199 @@
+package dev.responsive.kafka.api;
+
+import dev.responsive.kafka.store.ResponsiveMaterialized;
+import dev.responsive.kafka.store.ResponsiveStoreBuilder;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.state.KeyValueBytesStoreSupplier;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.TimestampedWindowStore;
+import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.WindowStore;
+
+/**
+ * A factory for creating Kafka Streams state stores on top of a Responsive storage backend.
+ * Use these when building your {@link org.apache.kafka.streams.Topology} to easily swap in
+ * Responsive stores wherever state is used.
+ *
+ * See {@link org.apache.kafka.streams.state.Stores} for instructions on how to plug in custom
+ * state stores and configure them.
+ */
+public final class ResponsiveStores {
```

Comment:
@agavra I tried to back out the API changes you took offense to so this PR isn't blocked on that discussion, but let me know if there's anything still in here you're uncomfortable with that I missed. The only change from the way it is today, other than being static, is to provide StoreBuilders (only) rather than StoreSuppliers. Either way they're both offered in `Stores` so I figured this would be ok?

Reply:
>  keeping the API with the supplier since we can use the supplier so we can eventually provide different implementations optimized for different use cases.

Personally I see this as an argument for moving towards a universal `KeyValueStoreOptions` rather than aligning to the existing Stores API...we can easily add new optional configurations that way, and it's easy to tell what's a variation on a fundamental store type vs different actual store types. This is part of what's so confusing about the existing `Stores` API and wild proliferation of different StoreSupplier static methods. 

Opens up so many questions...when can I use which store supplier? I mean just focusing on one aspect there's all this (not even getting into the new `versioned` store variants starting to roll out):
1. Are `#persistentKeyValueStore` and `#persistentTimestampedKeyValueStore` interchangeable? 
2. Are `#persistentKeyValueStore` and `#inMemoryKeyValueStore` interchangeable? 
3. Are `#persistentTimestampedKeyValueStore` and `#inMemoryKeyValueStore` interchangeable?

Want to guess? The answer may surprise you 😜

Reply:
Answer:
1. NEVER! Whether using the DSL or PAPI, you absolutely have to use the correct one -- this means somehow knowing to use the timestamped version for correct semantics in the DSL (ie via Materialized), and making sure to use the correct supplier API _only_ with its the corresponding builder API 
2. ALWAYS! Whether using the DSL or PAPI, you can actually swap out a `#persistentTimestampedKeyValueStore` for a `#inMemoryKeyValueStore` at any time or place in the code, or vice versa, it's all compatible
3. SOMETIMES! You can pass an `#inMemoryKeyValueStore` into either the timestamped or regular builder in the PAPI, but can't swap in a regular `#persistentKeyValueStore` to the timestamped builder. In the DSL, changing between these two will give you different semantics 😱

Reply:
ah... yikes 😨

Reply:
(This is why I'm leaning so heavily into providing only builder and materialized APIs for each type of store -- not only is there no actual need for the direct store suppliers*, it's a nightmare trying to figure out which one can/should be passed into which builder/materialized. Because the runtime is opinionated about this, but the `Stores` API is not)

* the one notable exception here being stream-stream  joins, but those are a nightmare to configure as well for a different reason, though the underlying cause is the same: the runtime is opinionated about there being correct values to the parameters, but the API makes this look the user's choice.